datumaro/plugins/coco_format/extractor.py

```diff
@@ -124,6 +161,55 @@ def _load_items(self, loader):
 
         return items
 
+    def _load_panoptic_items(self, config, panoptic_images):
+        import numpy as np
+        import PIL.Image
+        from collections import defaultdict
+        items = OrderedDict()
+
+        def rgb2id(color):
+            if isinstance(color, np.ndarray) and len(color.shape) == 3:
+                if color.dtype == np.uint8:
+                    color = color.astype(np.int32)
+                return color[:, :, 0] + 256 * color[:, :, 1] + 256 * 256 * color[:, :, 2]
+            return int(color[0] + 256 * color[1] + 256 * 256 * color[2])
+
+        imgs_info = {}
+        for img in config['images']:
+            imgs_info[img['id']] = img
+
+        for ann in config['annotations']:
+            img_id = ann['image_id']
+            image_path = osp.join(self._images_dir, imgs_info[img_id]['file_name'])
+            image_size = (imgs_info[img_id].get('height'), imgs_info[img_id].get('width'))
+            image = Image(path=image_path, size=image_size)
+            anns = list()
+
+            panoptic_image_path = osp.join(panoptic_images, ann['file_name'])
+            panoptic_format = np.array(PIL.Image.open(panoptic_image_path), dtype=np.uint32)
+            pan = rgb2id(panoptic_format)
+            rle_per_category = defaultdict(list)
+            for segm_info in ann['segments_info']:
+                cat_id = segm_info['category_id']
+                segm_id = segm_info['id']
+                mask = pan == segm_id
```

Comment:
Consider using the Mask class with a single shared mask image in memory and `lambda m, c: m == c` ([example](https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/plugins/voc_format/extractor.py#L310)) as a lazy_mask extractor to avoid memory explosion on mask splitting for large number of instances and pixels.